### PR TITLE
Get type code of the class from class name

### PR DESCRIPTION
**Pull request by isuruf**
Came up in symengine/SymEngine.jl#17.

### CODE REVIEW

File: symengine/cwrapper.cpp [hunk 1]
```diff
@@ -15,6 +15,9 @@
 #include <symengine/visitor.h>
 #include <symengine/printer.h>
 
+#define xstr(s) str(s)
+#define str(s) #s
```
Comment (abinashmeher999) [inline symengine/cwrapper.cpp]: Shouldn't `#define xstr(s) #s` work too?

Comment (isuruf) [inline symengine/cwrapper.cpp]: No, that wouldn't expand `s` if `s` is a macro

**Comment by jverzani** [comment]
As I think about this, it would be much more helpful to get the entire SYMENGINE_ENUM rather than being able to dynamically find the type name from an object. (This could be as simple as just exposing the size of the enum, so we could iterate on the julia side.) This is because with the approach in this pull request we can dynamically generate types as they come up, but this makes it hard to write methods to dispatch on these types, as they won't be available when the methods are compiled. To work around this, it seems we would have to trigger the method compilation after the type compilation. Not impossible, but much more cumbersome.

**Comment by isuruf** [comment]
From the Julia side, we can maintain a list of classes and then use this method to get their enums. Only a subset of the classes are needed, because we expose only a few functionalities.
For ex. RealDouble is not exposed to Julia yet and is not needed.

**Comment by jverzani** [comment]
Strike that last comment of mine, it was easy to work around by using parametric types. I'll push that change to my branch in a bit.

**Comment by isuruf** [comment]
This is ready for review. @abinashmeher999, @jverzani, can you review?

**Comment by jverzani** [comment]
It would be very useful as is.

**Pull request merged**